sim/midas/src/main/cc/rtlsim/Makefrag-verilator

```diff
@@ -15,7 +15,24 @@
 #   (optional) verilator_conf: An verilator configuration file
 
 VERILATOR ?= verilator --cc --exe
-override VERILATOR_FLAGS := --assert -Wno-STMTDLY -O3 \
+ifeq ($(shell grep -i "module.*ariane" $(emul_v)),)
+    verilator_assert = --assert
+endif
+override VERILATOR_FLAGS := \
+	-Wno-STMTDLY \
```

Comment:
These flags are added to make Ariane work. See the following (https://github.com/pulp-platform/ariane/blob/382865dbdcb135c7e3e4305d6e0cd3a996fda3d8/Makefile#L378-L387). Specifically, they do the following:

```
PINMISSING
Warns that a module has a pin which is not mentioned in a cell instantiation. If a pin is not missing it should still be specified on the cell declaration with a empty connection, using "(.pin_name())".

IMPLICIT
Warns that a wire is being implicitly declared (it is a single bit wide output from a sub-module.) While legal in Verilog, implicit declarations only work for single bit wide signals (not buses), do not allow using a signal before it is implicitly declared by a cell, and can lead to dangling nets.

fatal
Don't error on warnings.

PINCONNECTEMPTY
Warns that a cell instantiation has a pin which is connected to .pin_name(), e.g. not another signal, but with an explicit mention of the pin. It may be desirable to disable PINCONNECTEMPTY, as this indicates intention to have a no-connect.

ASSIGNDLY
Warns that you have an assignment statement with a delayed time in front of it.

DECLFILENAME
Warns that a module or other declaration's name doesn't match the filename with path and extension stripped that it is declared in. 

UNUSED
Warns that the specified signal is never used/consumed

UNOPTFLAT
Warns that due to some construct, optimization of the specified signal is disabled.

BLKANDNBLK
BLKANDNBLK is an error that a variable comes from a mix of blocked and non-blocking assignments.

-Wno-style
Disable all code style related warning messages
```

Reply:
I think we can either choose `Wno-style` or `Wall` (enable all style warnings). It depends on which you all prefer.

Reply:
I'd prefer this be exposed as a hook and the additional flags provided specially in ariane case.

However, for now could you add a comment explaining which flags you added and why (for ariane support) directly in the source.

Reply:
As per Wno-style, i personally don't care. Whatever was more consistent with the old behavior.

Reply:
Can you clarify what you mean by "hook"? Do you want an `ifdef` like statement to add the flags if Ariane is found? Or do you want something else?

Reply:
Discussed offline: Just have the minimal set of new flags with explanations.